docs/weave-tutorial/05-messages.md

```diff
@@ -0,0 +1,109 @@
+title: Messages
+sidebar_label: Messages
+---
+
+> [PR#2](https://github.com/iov-one/tutorial/pull/2): _Create msgs_
+
+Messages are requests for a change in the state, the action part of a transaction. They also need to be persisted (to be sent over the wire and stored on the blockchain), and must also be validated. They later are passed into [Handlers](https://godoc.org/github.com/iov-one/weave#Handler) to be processed and effect change in the blockchain state.
+
+## Messages vs. Transactions
+
+A message is a request to make change and this is the basic element of a blockchain. A transaction contains a message along with metadata and authorization information, such as fees, signatures, nonces, and time-to-live.
+
+A [Transaction](https://godoc.org/github.com/iov-one/weave#Tx) is fundamentally defined as anything persistent that holds a message:
+
+```go
+type Tx interface {
+    Persistent
+    // GetMsg returns the action we wish to communicate
+    GetMsg() (Msg, error)
+}
+```
+
+And every application can extend it with additional functionality, such as [Signatures](https://godoc.org/github.com/iov-one/weave/x/sigs#SignedTx), [Fees](https://godoc.org/github.com/iov-one/weave/x/cash#FeeTx), or anything else your application needs. The data placed in the Transaction is meant to be anything that applies to all modules, and is processed by a **Middleware**.
+
+A [Message](https://godoc.org/github.com/iov-one/weave#Msg) is also persistent and can be pretty much anything that an extension defines, as it also defines the Handler to process it. The only necessary feature of a Message is that it can return a `Path() string` which allows us to route it to the proper Handler.
+
+When we define a concrete transaction type for one application, we define it in protobuf with a set of possible messages that it can contain. Every application can add optional field to the transaction and allow a different set of messages, and the Handlers and Decorators work orthogonally to this, regardless of the **concrete** Transaction type.
+
+## Defining Messages
+
+Messages are similar to the `POST` endpoints in a typical API. They are the only way to effect a change in the system. Ignoring the issue of authentication and rate limitation, which is handled by the Decorators / Middleware, when we design Messages, we focus on all possible state transitions and the information they need to proceed.
+
+In the Orderbook example, we can imagine:
+
+- Create market
+- Create order book
+- Create order
+
+## Dive into Code
+
+First create your `msg.go` file. This is where the magic will happen.
+
+First create an __init()__ function and register to migration schema:
+
+```go
+func init() {
+    // Migration needs to be registered for every message introduced in the codec.
+    migration.MustRegister(1, &CreateOrderBookMsg{}, migration.NoModification)
+}
+```
+
+Define path that will be used for routing messages to handler:
+
+```go
+func (CreateOrderBookMsg) Path() string {
```

Comment:
You can add a comment `// Path implementes weave.Msg interface.` This may seem unnecessary if you know weave. I think it is helpful for beginners to understand why this method exists.